data/tools/check_mixed_indent

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+# encoding: utf-8
+"""
+Flag an error if a single file uses both tabs and spaces for indenting.
+
+This does not change files, on the assumption that the mess is better
+handled with a text editor or a code reformatting tool.
+"""
+
+import  argparse, codecs, re, sys
+
+def check_indent(file):
+    has_tabs = False
+    with codecs.open(file, "r", "utf8") as lines:
```

Comment:
The regular `open` function allows to specify an encoding. In this case, it should be enough and the `codecs` module could be removed from the script.

Reply:
Will change to regular `open`.

I found some new behavior when testing after changing to the regular `open`, but it must have been there when using `codecs` too. Each subsequent `for line in lines:` needs a `lines.seek(0)` first, otherwise it continues from the place that the previous `for` had left it in. With that in, found 1 file that had tabs and spaces on the same line.